frontend/src/SettingsDashboard.jsx

```diff
@@ -145,20 +160,32 @@ const SettingsTable = ({ name, settings, settingsType, onEditCell }) => {
         },
     ];
 
+    let settingsModal = null;
+    if (modalOpen) {
+        settingsModal = <SettingsModal
+            type={settingsType}
+            onClose={handleModalClose}
+            action={modalAction}
+            setting={modalData.setting}
+            value={modalData.value}
+            open={true}
```

Comment:
`open` is not a prop here any more

Reply:
It has to be present according to `material-ui` :/ I think the idea is that the modal object always exists as a single instance, rather than being created on demand, but it makes managing it's state much more complex (for no gain?).